### ISSUE TITLE
[Feature Request] IoC Container

### ISSUE THREAD

**Issue by shopblocks** [UnionOfRAD/lithium#1257]
Since dependency injection can be done via $_classes, this is helpful in methods for doing static::_instance('ClassName') to get an instance of the class you want, so that tests can override certain methods. 

An IoC Container would be a welcome addition, so that Instances can be instantiated in the parameter list rather than doing the call to _instance() ourselves.

Is this something that would be considered for the roadmap?

**Comment by fedeisas** [comment]
We're using [Pimple](https://github.com/silexphp/Pimple) in our li3 app and I've found really hard to inject controllers. It would be great if li3 had bindings to implement whatever IoC container you want.

**Comment by fedeisas** [comment]
Also, the documentation for the current implementation is sparse: http://li3.me/docs/manual/architecture/dependencies.md

**Comment by davidpersson** [comment]
@fedeisas Yes it's sparse, as sparse can get. Please open a ticket over at https://github.com/UnionOfRAD/manual so we have something to track it. Needless to say that we'd be glad for help expanding it ;)

**Comment by fedeisas** [comment]
Hi @davidpersson! I'd gladly write it, but I have to say I don't understand how this injection is supposed to work.

The `$_classes` and `$_autoConfig` variables of `lithium\core\Object` are really confusing to me. But hey, maybe it's just me.

Anyway, it would be great to have working examples on how to use those features.

**Comment by davidpersson** [comment]
I can shed some light on it :)
